packages/flutter/test/material/app_bar_test.dart

```diff
@@ -926,6 +927,9 @@ void main() {
     expect(getAppBarWidget(appBarFinder).elevation, 0.0);
 
     await tester.pumpWidget(buildSliverAppBar(true));
+    expect(getAppBarWidget(appBarFinder).elevation, null);
+
+    await tester.pumpWidget(buildSliverAppBar(true, 4.0));
     expect(getAppBarWidget(appBarFinder).elevation, 4.0);
   });
```

Comment:
@NWalker1208 I think it's better if you add a new test, maybe called 'SliverAppBar uses elevation from AppBarTheme' where you set the custom elevation etc. Then you actually test that the Material (`getMaterialWidget`) has an elevation, rather then the appBarWidget. You can keep the changes to this test.

Add a comment that it is a regression test for your issue. Otherwise it LGTM, I would ping someone from the Flutter team to review it.

Reply:
Thanks for the suggestion, I've added the test